x-pack/plugins/observability/public/pages/alerts/alerts.tsx

```diff
@@ -62,11 +63,22 @@ function InternalAlertsPage() {
     },
   } = useKibana<ObservabilityAppServices>().services;
   const { ObservabilityPageTemplate, observabilityRuleTypeRegistry } = usePluginContext();
-  const alertSearchBarStateProps = useAlertSearchBarStateContainer(URL_STORAGE_KEY);
+  const alertSearchBarStateProps = useAlertSearchBarStateContainer(URL_STORAGE_KEY, {
+    replace: false,
+  });
 
-  const chartThemes = {
+  const onBrushEnd: BrushEndListener = (brushEvent) => {
+    const { x } = brushEvent as XYBrushEvent;
+    if (x) {
+      const [start, end] = x;
+      alertSearchBarStateProps.onRangeFromChange(new Date(start).toISOString());
```

Comment:
Should we handle a case when x is just []. So `if (x)` will go with true. But `new Date(start).toISOString()` will throw the error if start is undefined.

Reply:
I don't know in which scenario this can happen, `XYBrushEvent` type is:
```
export interface XYBrushEvent {
    x?: [number, number];
    y?: Array<GroupBrushExtent>;
}
```
So either we don't have `x`, or when we have it, it has 2 numbers. I also didn't see such a check in other implementations of `onBrushEnd`.